source/docs/zero-to-robot/step-4/creating-test-drivetrain-program-python.rst

```diff
@@ -0,0 +1,153 @@
+Creating your Test Drivetrain Program (Python)
+==============================================
+
+Once everything is installed, we're ready to create a robot program. RobotPy comes with several templates for robot programs.  Use of these templates is highly recommended for new users; however, advanced users are free to write their own robot code from scratch. This article walks through creating a project from one of the provided examples which has some code already written to drive a basic robot.
+
+.. important:: This guide includes code examples that involve vendor hardware for the convenience of the user. In this document, PWM refers to the motor controller included in the KOP. The CTRE tab references the Talon FX motor controller (Falcon 500 motor), but usage is similar for TalonSRX and VictorSPX. The REV tab references the CAN SPARK MAX controlling a brushless motor, but it's similar for brushed motor. Using the :guilabel:`pip install robotpy[all]` during installation includes all vendor Python modules. There is an assumption that the user has already configured the device(s) (update firmware, assign CAN IDs, etc) according to the manufacturer documentation (`CTRE <https://docs.ctr-electronics.com/>`__ `REV <https://docs.revrobotics.com/sparkmax/gs-sm>`__).
+
+Creating a New Project
+----------------------
+
+Create a new folder and in this folder create a file named :guilabel:`robot.py`.
+Open Visual Studio Code and from the File menu and click on Open Folder. From there, select and open the newly created folder.
+
+If you start from a `template <https://github.com/robotpy/examples>`__, replace the robot.py file with the template one and add all other files.
+
+VS Code might show you this window:
+
+.. image:: /docs/software/vscode-overview/images/creating-robot-program/trusted-workspace.png
+   :alt: Trusted Workspace dialog in VS Code.
+
+Click :guilabel:`Yes I trust the authors`.
+
+Once opened we will see the project hierarchy on the left. Double clicking on the file will open that file in the editor.
+
+
+Basic code structure
+--------------------
+
+First, here is what a simple code can look like.
+
+.. tabs::
+
+   .. group-tab:: PWM
+
+         .. remoteliteralinclude:: https://raw.githubusercontent.com/robotpy/examples/main/getting-started/robot.py
```

Comment:
the RLIs should use a tag (preferred) or a commit hash, rather then main, so that new commits to main don't silently break the docs.